pkgs/applications/window-managers/xborders/default.nix

```diff
@@ -0,0 +1,39 @@
+{ lib, python3Packages, fetchFromGitHub, libwnck, gtk3, libnotify, wrapGAppsHook, gobject-introspection }:
+
+with python3Packages; buildPythonPackage rec {
+  pname = "xborders";
+  version = "3.4"; # in version.txt
+  src = fetchFromGitHub {
+    owner = "deter0";
+    repo = "xborder";
+    rev = "d840ed649e92b7ad399cdb1a30aeca7d09f2c08f";
+    sha256 = "sha256-Q2WWXekUL/OOr/OBqJOZbAHaWURbP+coBVp206uPSNY=";
+  };
+  buildInputs = [ libwnck gtk3 libnotify ];
+  nativeBuildInputs = [ wrapGAppsHook gobject-introspection ];
+  propagatedBuildInputs = [ pycairo requests pygobject3 ];
+  preBuild = ''
+    sed '52,58!d' ${pname}
+    sed -i '51 a\    return ${version}' ${pname}
```

Comment:
Do not abuse `${pname}`. Indeed, what these lines do?

Reply:
I mentioned what these do in change description.

> In `preBuild`, I'm replacing the contents of the `get_version` function with the version number, because otherwise xborders requires a version.txt file to be next to the executable to fetch the version number from.

Basically, the author does this really weird thing where the version number is stored in [version.txt](https://github.com/deter0/xborder/blob/d840ed649e92b7ad399cdb1a30aeca7d09f2c08f/version.txt#L1) and then is fetched in the [`get_version` method](https://github.com/deter0/xborder/blob/d840ed649e92b7ad399cdb1a30aeca7d09f2c08f/xborders#L51-L58). version.txt has to be next to the executable Python file, which is problematic since I don't think it'd be a good idea to have a data file in the `$out/bin` folder. It might also cause issues with the wrapping, I'm not sure.

All these lines do is replace the contents of `get_version` with a single return, removing the dependency of version.txt:

```PY
def get_version():
    return 3.4
```

Reply:
Well, the first obvious problem is it is strongly depends on the position of the code.
If by any reason the upstream modifies the lines, your sed script will destroy the file without warning.

Change this to an explicit `.diff` file instead.